pypeit/spectrographs/keck_mosfire.py

```diff
@@ -432,3 +516,233 @@ def tweak_standard(self, wave_in, counts_in, counts_ivar_in, gpm_in, meta_table,
+        self.slitmask = SlitMask(np.array([np.zeros(ssl['Slit_Number'].size),   # mosfire maskdef has not slit corners
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size),
+                                           np.zeros(ssl['Slit_Number'].size)]).T.reshape(-1,4,2),
+                                 slitid=np.array(ssl['Slit_Number'], dtype=int),
+                                 align=ssl['Target_Name'] == 'posB',
+                                 science=ssl['Target_Name'] != 'posB',
+                                 onsky=np.array([slit_ra,
+                                                 slit_dec,
+                                                 np.array(ssl['Slit_length'], dtype=float),
+                                                 np.array(ssl['Slit_width'], dtype=float),
+                                                 np.array([round(hdu[0].header['SKYPA3'],2)]*ssl['Slit_Number'].size)]).T,
+                                 objects=objects,
+                                 posx_pa=posx_pa)
+        return self.slitmask
+
+    def get_maskdef_slitedges(self, ccdnum=None, filename=None, debug=None):
+        """
+         Provides the slit edges positions predicted by the slitmask design using
+         the mask coordinates already converted from mm to pixels by the method
+        `mask_to_pixel_coordinates`.
+
+        If not already instantiated, the :attr:`slitmask`, :attr:`amap`,
+        and :attr:`bmap` attributes are instantiated.  If so, a file must be provided.
+
+        Args:
+        ccdnum (:obj:`int`):
+            Detector number
+        filename (:obj:`str`, optional):
+            The filename to use to (re)instantiate the :attr:`slitmask` and :attr:`grating`.
+            Default is None, i.e., to use previously instantiated attributes.
+        debug (:obj:`bool`, optional):
+            Run in debug mode
+        Returns:
+            Three `numpy.ndarray`_ and a :class:`~pypeit.spectrographs.slitmask.SlitMask`.
+            Two arrays are the predictions of the slit edges from the slitmask design and
+            one contains the indices to order the slits from left to right in the PypeIt orientation
+
+        """
+        # Re-initiate slitmask and amap and bmap
+        if filename is not None:
+            # Reset the slitmask
+            self.get_slitmask(filename)
+
+        if self.slitmask is None:
+            msgs.error('Unable to read slitmask design info. Provide a file.')
+
+        # Hard-Coded for MOSFIRE
+        platescale = 0.1798
```

Comment:
is this in the detector object?  If so, can we pull it from there?

Reply:
or follow what you do below (read from header)

Reply:
Done.